airflow/operators/python.py

```diff
@@ -145,44 +146,79 @@ def execute_callable(self):
         return self.python_callable(*self.op_args, **self.op_kwargs)
 
 
-class PythonFunctionalOperator(PythonOperator):
+class PythonFunctionalOperator(BaseOperator):
     """
-    Allows a workflow to "branch" or follow a path following the execution
-    of this task.
+    Wraps a Python callable and captures args/kwargs when called for execution.
 
-    It derives the PythonOperator and expects a Python function that returns
-    a single task_id or list of task_ids to follow. The task_id(s) returned
-    should point to a task directly downstream from {self}. All other "branches"
-    or directly downstream tasks are marked with a state of ``skipped`` so that
-    these paths can't move forward. The ``skipped`` states are propagated
-    downstream to allow for the DAG state to fill up and the DAG run's state
-    to be inferred.
+    .. seealso::
+        TBD
+
+    :param python_callable: A reference to an object that is callable
+    :type python_callable: python callable
+    :param multiple_outputs: if set, function return value will be 
+        unrolled to multiple XCom values. List will unroll to xcom values 
+        with index as key. Dict will unroll to xcom values with keys as keys. 
+        Defaults to False.
+    :type multiple_outputs: bool
     """
 
+    template_fields = ('_op_args', '_op_kwargs')
+    ui_color = '#ffefeb'
+
+    # since we won't mutate the arguments, we should just do the shallow copy
+    # there are some cases we can't deepcopy the objects(e.g protobuf).
+    shallow_copy_attrs = ('python_callable',)
+
+
     @apply_defaults
     def __init__(
         self,
+        python_callable: Callable,
         multiple_outputs: bool = False,
         *args,
         **kwargs
     ) -> None:
         super().__init__(*args, **kwargs)
+        if not callable(python_callable):
+            raise AirflowException('`python_callable` param must be callable')
+        self.python_callable = python_callable
         self.multiple_outputs = multiple_outputs
+        self._kwargs = kwargs
+        self._op_args = None
+        self._op_kwargs = None
```

Comment:
Should we add those to constructor?

Reply:
Given that the intended usage of the operator is with the `__call__` I would say we can keep those out from there. We have PythonOperator to serve those needs. Not strongly biased though, I thought it made sense to hide them a bit and constraint the usage of the operator a bit more.

I might even wonder if we should make this a private class instead so that it's only used with the task decorator. Thoughts?